Packages/SmithyTestUtil/Sources/JSONComparator.swift

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+public struct JSONComparator {
+    /// Returns true if the JSON documents, for the corresponding data objects, are equal.
+    /// - Parameters:
+    ///   - dataA: The first data object to compare to the second data object.
+    ///   - dataB: The second data object to compare to the first data object.
+    /// - Returns: Returns true if the JSON documents, for the corresponding data objects, are equal.
+    public static func jsonData(_ dataA: Data, isEqualTo dataB: Data) throws -> Bool {
+        let jsonDictA = try JSONSerialization.jsonObject(with: dataA)
+        let jsonDictB = try JSONSerialization.jsonObject(with: dataB)
+        return anyValuesAreEqual(jsonDictA, jsonDictB)
+    }
+}
+
+fileprivate func anyDictsAreEqual(_ lhs: [String: Any], _ rhs: [String: Any]) -> Bool {
+    guard lhs.keys == rhs.keys else { return false }
```

Comment:
There is no order here, since `keys` is a struct. I imagine the sequence and collection conformance methods likely will return keys in a non-deterministic order but those aren't in play here.

Reply:
Here's the relevant bits of the equatable conformance for the Keys struct https://github.com/apple/swift/blob/main/stdlib/public/core/Dictionary.swift#L1388-L1400